index.js

```diff
@@ -49,25 +58,45 @@ module.exports = function(opts) {
 
 };
 
+module.exports.getDefaultOptions = function (opts) {
+  opts = opts || {};
+  Object.keys(defaultOptions).forEach(function (optionName) {
+    if (opts[optionName] === undefined) {
+      opts[optionName] = defaultOptions[optionName];
+    }
+  });
+  return opts;
+};
+
 
-module.exports.getTplPath = function(req, opts) {
+module.exports.getTplPath = function(requestUrl, opts) {
+  opts = module.exports.getDefaultOptions(opts);
   assert(opts.ext && Array.isArray(opts.ext), 'opts.ext should be provided and be an array');
   assert(opts.baseDir, 'opts.baseDir should be provided');
 
-  var parsed = url.parse(req.originalUrl);
+  var parsed = url.parse(requestUrl);
   var pathname = parsed.pathname.replace(opts.baseUrl, '');
 
-  for (var i = 0; i < opts.ext.length; i++) {
-    var ext = opts.ext[i];
-    var filepath = path.join(opts.baseDir, pathname.replace(/\.html$/, ext));
+  if (opts.serveIndex && pathname.substr(-1) === '/') {
+    pathname += 'index' + opts.allowedExt[0];
+  }
+
+  var requestedExt = path.extname(pathname);
+  var requestedFilename = pathname.substr(0, pathname.length - requestedExt.length);
```

Comment:
```
requestedFilename = path.basename(pathname, requestedExt);
```

Should do a trick as well http://nodejs.org/api/path.html#path_path_basename_p_ext